### ISSUE TITLE
Add runkit php extension.

### ISSUE THREAD

**Issue by klonos** [WPN-XM/WPN-XM#498]
http://pecl.php.net/runkit

The extension was abandoned/neglected by the original author (Sara Golemon) due to lack of time/interest for years (since 6/6/2006 - version 0.9). Dmitry Zenovich was maintaining a fork of the extension in https://github.com/zenovich/runkit and Sara was kind enough to hand over her "baby" to him. This week, she added him as a maintainer of the project (history of the whole thing in https://github.com/zenovich/runkit/issues/89).

This repo contains windows .dll versions of the extension based on Dmitry's fork (not a fork any more): https://github.com/Crack/runkit-windows
##

**Comment by jakoch** [comment]
Runkit is quite useful, when working with online editors to restrict the eval() possibilities and to build sandboxed environments.

What a story... its a good example for the flawed process of transfering ownership of PECL extensions. Its simpler for people to fork an extension and do unofficial releases than to get patches integrated or even get into a maintainer or dev role. Maybe after a decade or so... :)
But its quite nice of Sara to contact Bjori and speeding the transfer up.

Lets wait for the official Windows builds for this extensions. 
They should show up on http://pecl.php.net/runkit 

**Todo**
- [ ] wait for release on http://pecl.php.net/runkit
- [ ] add crawler
- [ ] add to registry
- [ ] add to download description file
- [ ] add to installers

**Comment by klonos** [comment]
Thanx for considering this @jakoch and yes, it was very noble of Sara to not only hand over the maintenance but also help with the process of getting Dmitry to become an "official" maintainer of the PECL repo.

Sorry, I had no idea that Windows .dlls are auto-created after some time - I thought that Dmitry would have to actually build them. That's why I pointed to the github repo that already had them.

**Comment by jakoch** [comment]
@zenovich Hey Dmitry, would you be so nice to upload the latest build for Windows to either PECL or Github Releases. Thank you,

**Comment by zenovich** [comment]
Hey @jakoch,
I want to do it on PECL, but I don't know how. Is there any instruction for adding Windows binaries to PECL?

**Comment by jakoch** [comment]
> Is there any instruction for adding Windows binaries to PECL?

I don't know. It seems to be undocumented, because there is no entry on the Wiki http://wiki.php.net/internals/windows nor in the docs - at least i could not find any information regarding "How to release Windows extension dlls on PECL".

I always thought, that the packages are automatically build for Windows once registered on PECL,
but it seems, that my thinking is wrong.

I've never seen the backend of PECL, but maybe its possible to login and upload packages there manually. And in case, that uploading works only for `tar.gz` packages, i would suggest to contact Pierre directly for getting the windows builds included - alternatively you could post to php-internals. 
The goal is that the DLLs show up over at http://windows.php.net/downloads/pecl/releases/

**Comment by zenovich** [comment]
Maybe there is some mechanism to build them from sources automatically, I receive an email about absense of the file config.w32 every time I'm doing release on PECL.

Nevertheless there should be the way to upload dlls manually, but I couldn't find it. I hope @pierrejoye can clarify this.

For now, you can download dlls from https://github.com/Crack/runkit-windows

**Comment by jakoch** [comment]
Yes, during upload to PECL the extension is checked for conformity.
Some files, like package.xml and some of its directives are expected to be there.

**Comment by jakoch** [comment]
Oh, look.. there is movement:  
- http://windows.php.net/downloads/pecl/releases/runkit/1.0.4/
- new folder runkit - date `12/7/2016` 
- 1.0.4	stable	2015-10-19

**Comment by klonos** [comment]
Well, it is required by a Drupal contrib module called [Extended Path Aliases](https://www.drupal.org/project/path_alias_xt). I have created [an issue in the queue](https://www.drupal.org/node/760758) providing pre-compiled binaries for various OSes and others have contributed as well. I use Ubuntu for my live sites and they either have up-to-date libraries or I know how to compile them myself. I use Windows for my local dev (and WPN-XM of course), so I needed that library in some cases. Hence the issue here 😄 

In the meantime, I have (mainly) moved to [Backdrop CMS](http://backdropcms.org) which is a fork of Drupal 7 that includes most of the modules I need related to path in core. I have also filed an issue to merge [Extended Path Aliases](https://github.com/backdrop/backdrop-issues/issues/1418) functionality in core and I trust it will happen soon.

So, to answer the question, yes, otherwise you might get the odd "WPN-XM doesn't work with Drupal" or "Drupal doesn't work on WPN-XM" depending on how many people use the module and how many are developing locally on Windows.
